graphdatascience/procedure_surface/api/wcc_endpoints.py

```diff
@@ -0,0 +1,276 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, List, Optional
+
+from pandas import DataFrame
+
+from ...graph.graph_object import Graph
+
+
+class WccEndpoints(ABC):
+    """
+    Abstract base class defining the API for the Weakly Connected Components (WCC) algorithm.
+    """
+
+    @abstractmethod
+    def mutate(
+        self,
+        G: Graph,
+        mutate_property: str,
+        threshold: Optional[float] = None,
+        relationship_types: Optional[List[str]] = None,
+        node_labels: Optional[List[str]] = None,
+        sudo: Optional[bool] = None,
+        log_progress: Optional[bool] = None,
+        username: Optional[str] = None,
+        concurrency: Optional[Any] = None,
+        job_id: Optional[Any] = None,
+        seed_property: Optional[str] = None,
+        consecutive_ids: Optional[bool] = None,
+        relationship_weight_property: Optional[str] = None,
+    ) -> WccMutateResult:
```

Comment:
This constitutes a breaking change. Before we would have returned a Dataframe. 
However I think this API is much easier to use as it is more strict on the types and should enabled better IDE autocompletion.

Reply:
i also like the typed api more :) 

especially if we also offer a to_pandas utility (using the from_dict utility).
mainly useful for display in jupyter notebooks

Reply:
I also prefer the typed API. I don't like breaking things, but I do like having a result type that we own and control, and especially if it comes with static fields to help the user.